api/turing/cluster/virtual_service_test.go

```diff
@@ -62,6 +60,8 @@ func TestBuildVirtualService(t *testing.T) {
 		},
 	}
 	got := cfg.BuildVirtualService()
-	err := tu.CompareObjects(expected, *got)
-	assert.NilError(t, err)
+
+	assert.Equal(t, expected.ObjectMeta, got.ObjectMeta)
+	assert.Equal(t, expected.Spec.String(), got.Spec.String())
+	assert.Equal(t, expected.Status.String(), got.Status.String())
```

Comment:
We cannot compare `expected` and `got` directly since `VirtualService`-s returned by the Istio client now contain unexported fields (e.g. `_.Status.state`) which makes `CompareObjects` unusable, as well as mutex locks. In order to work around this, we can compare the string generated from the `VirtualService`'s spec and status that prints out most of the fields in these fields.

The assert package has also been changed to `github.com/stretchr/testify/assert` to make the testing packages consistent with what is found in `api/turing/cluster/controller_test.go`. The previous `gotest.tools/assert` package was also unable to properly compare the `ObjectMeta` fields:

```
panic: runtime error: comparing uncomparable type v1.ObjectMeta
```